src/java.base/share/classes/sun/security/util/DisabledAlgorithmConstraints.java

```diff
@@ -130,21 +130,15 @@ public DisabledAlgorithmConstraints(String propertyName,
         disabledAlgorithms = getAlgorithms(propertyName);
 
         // Check for alias
-        int ecindex = -1, i = 0;
+        Pattern INCLUDE_PATTERN = Pattern.compile(
+                "include " + PROPERTY_DISABLED_EC_CURVES);
         for (String s : disabledAlgorithms) {
-            if (s.regionMatches(true, 0,"include ", 0, 8)) {
-                if (s.regionMatches(true, 8, PROPERTY_DISABLED_EC_CURVES, 0,
-                        PROPERTY_DISABLED_EC_CURVES.length())) {
-                    ecindex = i;
-                    break;
-                }
+            Matcher matcher;
+            if ((matcher = INCLUDE_PATTERN.matcher(s)).matches()) {
+                disabledAlgorithms.remove(matcher.group());
+                disabledAlgorithms.addAll(getAlgorithms(PROPERTY_DISABLED_EC_CURVES));
```

Comment:
It may increase the readability a little bit if having the assignment in the declaration line:
```
-             Matcher matcher;
-             if ((matcher = INCLUDE_PATTERN.matcher(s)).matches()) {
+             Matcher matcher = INCLUDE_PATTERN.matcher(s);
+             if (matcher.matches()) {
```